crates/lakekeeper/src/service/storage/s3.rs

```diff
@@ -747,6 +754,24 @@ impl S3Profile {
             assume_role_builder
         };
 
+        let assume_role_builder = if self.sts_session_tags.is_empty() {
```

Comment:
I had to swap the ordering on this, but I verified the session tags are working with the latest code